src/main/java/olya/app/remindme/controller/UserController.java

```diff
@@ -22,25 +28,74 @@
 public class UserController {
     private final UserService userService;
     private final UserMapper userMapper;
+    private final TokenService tokenService;
 
+    //public
     @PostMapping("/register")
     public UserResponseDto register(@Valid @RequestBody UserRequestDto userRequestDto) {
         User user = userService.create(userRequestDto);
         return userMapper.mapToDto(user);
     }
 
+    //for user and admin
+    @GetMapping("/my-user")
+    public UserResponseDto getCurrentUser(@AuthenticationPrincipal UserPrincipal principal) {
+        User user = userService.getByEmail(principal.getEmail());
+        return userMapper.mapToDto(user);
+    }
+
+    //for user and admin
+    @PutMapping("/my-user")
+    public UserResponseDto updateCurrentUser(@AuthenticationPrincipal UserPrincipal principal, @Valid @RequestBody UserRequestDto userRequestDto) {
+        User user = userService.getByEmail(principal.getEmail());
+        User updatedUser = userService.update(user, userRequestDto);
+        return userMapper.mapToDto(updatedUser);
+    }
+
+    //for user and admin
+    @PutMapping("/my-user/deactivate")
```

Comment:
Should user be able to deactivate himself?

Reply:
yes